src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.epics.js

```diff
@@ -28,13 +29,14 @@ const deriveSelectedName = (attributes = {}) => attributes.reduce((acc, { value:
 
 const fetchTeiStream = teiId => from(fetchTrackedEntityInstance(teiId))
     .pipe(
-        map(({ attributes, enrollments }) => {
-            // todo this is not scaling when you have many attributes the name will be huge
-            const selectedName = deriveSelectedName(attributes);
+        map(({ attributes, enrollments, trackedEntityType }) => {
+            const teiDisplayName = deriveSelectedName(attributes);
             const enrollmentsSortedByDate = sortByDate(enrollments);
+            const { name: tetDisplayName } = getTrackedEntityTypeThrowIfNotFound(trackedEntityType);
```

Comment:
I would prefer if we derived this from the program metadata when rendering the page instead of saving it to the redux store. What do you think?

Reply:
Yep lets tackle this here. You ve made another comment about it. But I think it is best if I deal with it here since its much higher up the chain. ⛓️